tests/v2/validation/configmaps/configmaps_test.go

```diff
@@ -0,0 +1,107 @@
+package configmaps
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+  "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+  "github.com/rancher/rancher/tests/framework/extensions/configmaps"
+)
+
+type ConfigmapTestSuite struct {
+  suite.Suite
+  client            *rancher.Client
+  session           *session.Session
+  cluster           *management.Cluster
+}
+
+func (cm *ConfigmapTestSuite) TearDownSuite() {
+  cm.session.Cleanup()
+}
+
+func (cm *ConfigmapTestSuite) SetupSuite() {
+  cm.session = session.NewSession()
+
+  client, err := rancher.NewClient("", cm.session)
+  require.NoError(cm.T(), err)
+  cm.client = client
+
+  clusterName := client.RancherConfig.ClusterName
+  require.NotEmptyf(cm.T(), clusterName, "Cluster name to install should be set")
+  clusterID, err := clusters.GetClusterIDByName(cm.client, clusterName)
+  require.NoError(cm.T(), err, "Error getting cluster ID")
+  cm.cluster, err = cm.client.Management.Cluster.ByID(clusterID)
+  require.NoError(cm.T(), err)
+}
+
+func (cm *ConfigmapTestSuite) TestConfigMapCreate() {
+  subSession := cm.session.NewSession()
+  defer subSession.Cleanup()
+
+  configMapClient := cm.client.Steve.SteveType(configmaps.ConfigMapSteveType)
+  
+  log.Info("Creating a configmap")
+  configmapName := namegenerator.AppendRandomString(cmName)
+  configmapObj, err := createConfigmap(*configMapClient, configmapName, namespace, nil, labels, data)
+  require.NoError(cm.T(), err)
+
+  log.Info("Validating configmap was created")
+  assert.Contains(cm.T(), configmapObj.Name, configmapName)
+}
+
+func (cm *ConfigmapTestSuite) TestConfigMapUpdate(){
+  subSession := cm.session.NewSession()
+  defer subSession.Cleanup()
+
+  configMapClient := cm.client.Steve.SteveType(configmaps.ConfigMapSteveType)
+
+  log.Info(("Creating a configmap"))
+  configmapName := namegenerator.AppendRandomString(cmName)
+  configmapObj, err := createConfigmap(*configMapClient, configmapName, namespace, nil, labels, data)
+  require.NoError(cm.T(), err)
+
+  log.Info("Updating the configmap")
+  updatedConfigMapObj, err := updateConfigmapAnnotations(*configMapClient, configmapObj, map[string]string{annoKey: annoVal, descKey: descVal})
+  require.NoError(cm.T(), err)
+
+  log.Info("Validating configmap was properly updated")
+  annotations := updatedConfigMapObj.ObjectMeta.Annotations
+  for anno := range annotations{
+    if _, found := annotations[anno]; found{
+      expectedAnnotations[anno] = annotations[anno]
```

Comment:
this itself doesn't validate anything. It only gets us the expected annotations as a map. We would need to add the assertions to make sure the expected labels match the actual labels. Here's an example for the same: https://github.com/rancher/rancher/blob/7846f19f3f2731d68101a7981c2de2cfa8fcefb8/tests/v2/validation/rbac/rbac_psa_test.go#L102C1-L103C47

We can add this as a function in configmaps.go file, then validate if annotations match here.